Modules/pyexpat.c

```
@@ -1632,147 +1632,188 @@ static int init_handler_descrs(void)
+        Py_DECREF(errors_module);
+        Py_CLEAR(errmod_name);
+        goto error;
+    }
+    Py_CLEAR(errmod_name);
+
+    PyObject *model_module = PyModule_New(MODULE_NAME ".model");
+    if (model_module == NULL) {
+        goto error;
+    }
+
+    PyObject *modelmod_name = PyUnicode_FromString(MODULE_NAME ".model");
+    if (modelmod_name == NULL) {
+        goto error;
     }
-    Py_DECREF(modelmod_name);
+
+    if (_PyImport_SetModule(modelmod_name, model_module) < 0) {
+        Py_DECREF(model_module);
+        Py_CLEAR(modelmod_name);
+        goto error;
+    }
+    /* gives away the reference to model_module */
+    if (PyModule_AddObject(mod, "model", model_module) < 0) {
+        Py_DECREF(model_module);
+        Py_CLEAR(modelmod_name);
+        goto error;
+    }
+    Py_CLEAR(modelmod_name);
+
     if (errors_module == NULL || model_module == NULL) {
         /* Don't core dump later! */
-        Py_DECREF(m);
-        return NULL;
+        goto error;
     }
 
 #if XML_COMBINED_VERSION > 19505
     {
         const XML_Feature *features = XML_GetFeatureList();
         PyObject *list = PyList_New(0);
-        if (list == NULL)
-            /* just ignore it */
-            PyErr_Clear();
+        if (list == NULL) {
+            goto error;
+        }
         else {
             int i = 0;
             for (; features[i].feature != XML_FEATURE_END; ++i) {
-                int ok;
                 PyObject *item = Py_BuildValue("si", features[i].name,
                                                features[i].value);
                 if (item == NULL) {
                     Py_DECREF(list);
-                    list = NULL;
-                    break;
+                    goto error;
                 }
-                ok = PyList_Append(list, item);
+                int ok = PyList_Append(list, item);
                 Py_DECREF(item);
                 if (ok < 0) {
                     PyErr_Clear();
```

I would prefer to not silently ignore errors.